### PR TITLE
Gh 327 event logging

### PR DESCRIPTION
**Pull request by rocky341**
#327 

Introduces event logging with admins able to access a h2 console in order to see event logs. 

Logs are persistent

### CODE REVIEW

**Comment by rocky341** [comment]
Looking into why semgrep is failing on the pipeline

**Comment by rocky341** [comment]
@t92549 @GCHQDeveloper314 

Fixed the semgrep issue looks like Semgrep received an update which changed the way things work a bit. Semgrep now needs its own yaml file and I have updated the settings to run and then upload findings to the security dashboard. The positive of this move is that for false positives such as the `middleware.js` error, which kept breaking the pipeline, can now be dismissed as a false positive however is still documented in the security dashboard.

**Comment by rocky341** [comment]
@t92549 @GCHQDeveloper314 

Ready for review

**Review by GCHQDeveloper314 — APPROVED** [review]

File: gaffer-as-a-service/deploy/helm/charts/gaas-rest/templates/deployment.yaml [hunk 1]
```diff
@@ -42,6 +42,9 @@ spec:
               limits:
                 memory: "1Gi"
                 cpu: "1000m"
+            volumeMounts:
```
[the remark this answers is not in the document]
Comment (rocky341) [reply gaffer-as-a-service/deploy/helm/charts/gaas-rest/templates/deployment.yaml:45]: No this is related to the persistent event logging. Its part of the set up to allow the h2 database to be persistent through restarts/failures etc.

[the remark this answers is not in the document]
Comment (t92549) [reply gaffer-as-a-service/deploy/helm/charts/gaas-rest/templates/deployment.yaml:45]: Ah I see it's 2 things in one, I asked for a ticket to fix semgrep and presumed this was it, never mind

**Review by t92549 — APPROVED** [review]

**Pull request merged**